components/script/dom/htmlelement.rs

```diff
@@ -278,6 +278,39 @@ fn to_snake_case(name: DOMString) -> DOMString {
     attr_name
 }
 
+/*
+ * https://html.spec.whatwg.org/multipage/#attr-data-
+ * if this attribute is in snake case with a data- prefix,
+ * this function returns a name converted to camel case
+ */
+fn to_camel_case(name: DOMString) -> Option<DOMString> {
+    let has_uppercase = name.chars().fold(false, |exists_rest, curr_char| {
```

Comment:
Would be a little nicer if you used any or all (http://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all)